### PR TITLE
improve logging and fix lint issues

### PR DESCRIPTION
**Pull request by dangula**

### CODE REVIEW

**Comment by dangula** [comment]
running into "no space left on devie" when stashing workspace

**Comment by bassam** [comment]
I think we should explore using a docke registry to push the images and delete them possibly at the end. 

Also I think you can move some of the logic you have in the Jenkins file to inside the make files. Let discuss

**Comment by travisn** [comment]
i would suggest moving lint to a separate PR. Perhaps other commits would have made sense in a separate PR as well, it's good to keep a PR focused.

File: tests/framework/installer/install_helper.go [hunk 1]
```diff
@@ -24,37 +24,47 @@ import (
...
 	"github.com/rook/rook/tests/framework/utils"
+	"strings"
 )
 
 const (
 	rookOperatorFileName   = "rook-operator.yaml"
 	rookClusterFileName    = "rook-cluster.yaml"
 	rookToolsFileName      = "rook-tools.yaml"
 	podSpecPath            = "src/github.com/rook/rook/demo/kubernetes"
+	podSpecPath1_5         = "src/github.com/rook/rook/demo/kubernetes/1.5"
```
Comment (travisn) [inline tests/framework/installer/install_helper.go]: to support multiple versions it would be good factor out some properties to a new type or use an interface so the tests can be version agnostic

File: tests/framework/installer/install_helper.go [hunk 2]
```diff
@@ -24,37 +24,47 @@ import (
...
 type InstallHelper struct {
 	transportClient contracts.ITransportClient
 	isRookInstalled bool
 }
 
-func getPodSpecPath() string {
-	return filepath.Join(os.Getenv("GOPATH"), podSpecPath)
+func getPodSpecPath(k8sverion string) string {
+	switch k8sverion {
+	case "v1.5":
```
Comment (travisn) [inline tests/framework/installer/install_helper.go]: switch statements like this wouldn't be needed with the previous suggestion

File: tests/framework/utils/k8s_helper.go [hunk 3]
```diff
@@ -395,4 +410,19 @@ func (k8sh *K8sHelper) WaitUntilPodIsDeleted(podNamePattern string) bool{
...
 	return false
-}
\ No newline at end of file
+}
+
+func (k8sh *K8sHelper) WaitUntilPVCIsBound(pvcname string) bool {
+
+	inc := 0
+	for inc < 30 {
+		out, _ := k8sh.GetPVCStatus(pvcname)
```
Comment (travisn) [inline tests/framework/utils/k8s_helper.go]: is there any error state we should watch for? Otherwise in an error condition we will have to wait for timeout after the full retry loop

File: tests/framework/utils/s3_operations.go [hunk 4]
```diff
@@ -24,12 +24,15 @@ import (
...
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/coreos/pkg/capnslog"
 )
 
 type S3Helper struct {
 	s3client *s3.S3
 }
 
+var s3logger = capnslog.NewPackageLogger("github.com/rook/rook", "s3utils")
```
Comment (travisn) [inline tests/framework/utils/s3_operations.go]: why have an s3-specific logger? if you need a specific error, you could include `s3` in the message.

File: tests/framework/utils/s3_operations.go [hunk 5]
```diff
@@ -57,6 +61,7 @@ func (h *S3Helper) CreateBucket(name string) (bool, error) {
 		Bucket: aws.String(name),
 	})
 	if err != nil {
+		s3logger.Errorf("error encouterd : %v", err)
```
Comment (travisn) [inline tests/framework/utils/s3_operations.go]: a more specific error message would be helpful. Also a typo: `encounterd`

File: tests/smoke/block_storage_smoke_test.go [hunk 6]
```diff
@@ -100,12 +104,12 @@ func (suite *SmokeSuite) TearDownSuite() {
...
 func retryBlockImageCountCheck(h *SmokeTestHelper, imageCount int, expectedChange int) bool {
 	inc := 0
-	for inc < 30 {
+	for inc < 60 {
 		blockImages, _ := h.GetBlockClient().BlockList()
 		if imageCount+expectedChange == len(blockImages) {
 			return true
 		} else {
-			time.Sleep(time.Second * 2)
+			time.Sleep(time.Second * 3)
```
Comment (travisn) [inline tests/smoke/block_storage_smoke_test.go]: again, would be good to have error checks so we don't wait for the timeout in error conditions

File: MAINTAINERS [hunk 7]
```diff
@@ -0,0 +1,5 @@
+Bassam Tabbara <bassam.tabbara@quantum.com> (@bassam)
```
Comment (bassam) [inline MAINTAINERS]: why is this here. You might need to rebase

File: Jenkinsfile [hunk 8]
```diff
@@ -29,6 +30,73 @@ pipeline {
                 sh 'build/run make -j\$(nproc) test-integration'
             }
         }
+
+        stage('Integration Tests SetUp') {
```
Comment (bassam) [inline Jenkinsfile]: can you combine this with parallel runs stage. we just need one "integration tests" stage

File: Jenkinsfile [hunk 9]
```diff
@@ -29,6 +30,73 @@ pipeline {
...
+            }
+            steps {
+                parallel(
+                    "GCE_K8s_v1.7" : {
+                        node("gce_vm"){
+                            script {
+                                try{
+                                    withEnv(["KUBE_VERSION=v1.7"]){
+                                        unstash 'repo'
+                                        deleteDir()
```
Comment (bassam) [inline Jenkinsfile]: I dont follow why we are deleting the workspace here.

File: Jenkinsfile [hunk 10]
```diff
@@ -29,6 +30,73 @@ pipeline {
...
+                                        sh "docker load -i rookamd64.tar"
```
Comment (bassam) [inline Jenkinsfile]: you are using `saveImages.sh` above. I think there needs to be an equivalent `load images.sh`

File: Jenkinsfile [hunk 11]
```diff
@@ -29,6 +30,73 @@ pipeline {
...
+                                        sh "docker load -i toolboxamd64.tar"
+                                        sh 'build/run make -j\$(nproc) build.common'
```
Comment (bassam) [inline Jenkinsfile]: why is a build needed here? should be a pre-req for test-integration in the makefile?

File: Jenkinsfile [hunk 12]
```diff
@@ -29,6 +30,73 @@ pipeline {
...
+                                        sh 'tests/scripts/kubeadm-dind.sh up'
+                                        sh 'build/run make -j\$(nproc) test-integration'
+                                    }
+                                }
+                                finally{
```
Comment (bassam) [inline Jenkinsfile]: given the we are using kernel rbd, and kernel cephfs its conceivable that we are going to leave state behind on the slave/agent that interferes with future tests. can we just each slave for "integration tests" once? i.e. restart it at the VM/cloud level? that would remove the need to cleanup here.

File: Jenkinsfile [hunk 13]
```diff
@@ -29,6 +30,73 @@ pipeline {
...
+            }
+        }
+
+        stage('Integration Tests - parallel runs') {
+            environment {
+                KUBE_VERSION = "v1.6"
+            }
+            steps {
+                parallel(
+                    "GCE_K8s_v1.7" : {
```
Comment (bassam) [inline Jenkinsfile]: can this section be written as a loop of some sort? lots of code duplication.

File: Jenkinsfile [hunk 14]
```diff
@@ -29,6 +30,73 @@ pipeline {
...
+        stage('Integration Tests SetUp') {
+            steps {
+                sh 'tests/scripts/saveImages.sh'
+                stash name: 'repo', useDefaultExcludes: false
+            }
+        }
+
+        stage('Integration Tests - parallel runs') {
+            environment {
+                KUBE_VERSION = "v1.6"
```
Comment (bassam) [inline Jenkinsfile]: is this needed still?

File: Jenkinsfile [hunk 15]
```diff
@@ -29,6 +30,73 @@ pipeline {
                 sh 'build/run make -j\$(nproc) test-integration'
             }
         }
+
+        stage('Integration Tests SetUp') {
+            steps {
+                sh 'tests/scripts/saveImages.sh'
+                stash name: 'repo', useDefaultExcludes: false
```
Comment (bassam) [inline Jenkinsfile]: should you exclude .cache, .workdir, _output and maybe even .git?

**Comment by bassam** [comment]
@dangula please rebase. lots of unnecessary changes here. will continue review after rebase.

File: tests/framework/utils/k8s_helper.go (hunk 3 above)
Comment (travisn) [inline tests/framework/utils/k8s_helper.go]: when we have retry loops like this it would be good to have log statements so we can tell what is happening. Long delays between logging makes troubleshooting difficult. That can be in another PR though

Comment (dangula) [inline tests/framework/utils/k8s_helper.go]: This file will be going through another refactor, i will add more logging then.

**Comment by bassam** [comment]
@dangula its quite hard to follow this PR. can you split the lint changes into a separate commit please? I'd like to review the Jenkinsfile/saveimages.sh changes at least.

**Comment by dangula** [comment]
@bassam  this PR is getting confusing so I have removed, the jenkinsfile/saveimages.sh from this pr, I will create another pr for issue #786.  shortly.

**Comment by bassam** [comment]
I still see the save image.sh files in the files changed.

**Comment by dangula** [comment]
missed that one, its removed now

**Comment by dangula** [comment]
reverted dind-cluster.sh
